### PR TITLE
Add voting for items

### PR DESCRIPTION
**Pull request by SpencerCornish**
Adds an upvote capability to items. 
TODO:
- add counter on items
- add down vote?
- add fun styling and effects to the upvote button

### CODE REVIEW

**Comment by jerold** [comment]
Your null warning was due to the actions being added to items.dart not having its auto-generated implementation added to items.g.dart. You'll need to run `dart tool/build.dart` after making changes to any of the state or model files (or any file ending .g.dart).

File: lib/src/components/itemCard/itemCard.html [hunk 1]
```diff
@@ -1,12 +1,9 @@
 <div class="box is-primary">
   <div class="columns is-mobile">
     <div class="column is-narrow">
-      <p>{{item.supporterUids}}</p>
-    </div>
-    <div class="column is-narrow">
-      <p class="control">
-        <a class="button is-success is-outlined is-small" (click)="upVote()"><i class="fa fa-arrow-up" aria-hidden="true"></i></a>
-      </p>
+      <span class="icon is-danger">
```
Comment (jerold) [inline lib/src/components/itemCard/itemCard.html]: ```is-danger``` class doesn't seem to add color to icons. :( shame.

**Comment by SpencerCornish** [comment]
Thanks for the help!

**Comment by jerold** [comment]
Lets change the value being saved on these votes from bool to int, and then merge. :) Nice work.

**Pull request merged**